spec-main/api-net-spec.ts

```diff
@@ -1626,6 +1631,137 @@ describe('net module', () => {
       await collectStreamBody(response);
     });
 
+    it('should not join repeated discardable header values with ,', async () => {
+      const serverUrl = await respondOnce.toSingleURL((request, response) => {
+        response.statusCode = 200;
+        response.statusMessage = 'OK';
+        response.setHeader('last-modified', ['yesterday', 'today']);
+        response.end();
+      });
+      const urlRequest = net.request(serverUrl);
+      const response = await getResponse(urlRequest);
+      expect(response.statusCode).to.equal(200);
+      expect(response.statusMessage).to.equal('OK');
+
+      const headers = response.headers;
```

Comment:
Let me know if it'd be better to have tests not so explicitly related to `IncomingMessage.rawHeaders` be a different PR that is merged before/after/concurrently to this.